index.js

```diff
@@ -1,8 +1,141 @@
-import { NativeModules, Platform } from 'react-native';
+import { NativeModules, NetInfo, Platform } from 'react-native';
 import ReactNativeFS from 'react-native-fs';
+
 const { RNAppUpgrade } = NativeModules;
 
+let jobId = -1;
+const DEFAULT_IOS_APP_LOOKUP_URL = 'https://itunes.apple.com/lookup?id=';
+
 class AppUpgrade {
+  constructor(options) {
+    this.options = options;
+  }
+
+  async _requestUrlAsync(method, url, params) {
+    let request = {};
+    if (method === 'POST') {
+      request = {
+        method: method,
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(params)
+      };
+    }
+
+    return new Promise((resolve, reject) => {
+      fetch(url, request)
+        .then(response => response.json())
+        .then(resolve)
+        .catch(reject);
+    });
+  }
+
+  get downloading() {
+    return jobId > -1;
+  }
+
+  progress(data) {
+    const percentage = (100 * data.bytesWritten / data.contentLength) | 0;
+    this.options.downloadApkProgress && this.options.downloadApkProgress(percentage);
+  }
+
+  begin() {
+    this.options.downloadApkStart && this.options.downloadApkStart();
+  }
+
+  _handleResult(data) {
+    if (data.resultCount < 1) {
+      return;
+    }
+    const result = data.results[0];
+    const version = result.version;
+    const trackViewUrl = result.trackViewUrl;
+    if (version !== RNAppUpgrade.versionName) {
+      if (this.options.needUpdateApp) {
+        this.options.needUpdateApp(isUpdate => {
+          if (isUpdate) {
+            RNAppUpgrade.installFromAppStore(trackViewUrl);
+          }
+        });
+      }
+    }
+  }
+
+  _handleError(err) {
+    console.log('downloadApkError', err);
+    this.options.onError && this.options.onError(err);
+  }
+
+
+  getNetworkStatus() {
+    return new Promise(async(resolve) => {
+      NetInfo.fetch().done(resolve);
+    })
+  }
+
+  getLocalVersion() {
+    const localVersionName = RNAppUpgrade.versionName;
+    const localVersionCode = RNAppUpgrade.versionCode;
+    return {localVersionName, localVersionCode};
+  }
+
+  /**
+   * update app and install
+   * @param apkUrl
+   */
+  updateAndroidApp(apkUrl) {
+    if (this.downloading) {
+      return;
+    }
+
+    const progressDivider = 1;
+    const downloadDestPath = this.options.downloadApkSavePath
+      ? this.options.downloadApkSavePath
+      : `${ReactNativeFS.DocumentDirectoryPath}/NewApp.apk`;
```

Comment:
Why not use `const a = xxx || 'default'`

Reply:
`NewApp.apk` update to `TempApp_${Date.now()}.apk`

Reply:
ok，good option

Reply:
I guess TempApp_${versionName}.apk may be better

Reply:
If use version to mark the file,the next time to check App upgrade, we can easily find is it to download apk again.